string data_string
string directory
string name_file
---
